nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -30,6 +46,39 @@ in
         description = "Extra flags passed to the GlusterFS daemon";
         default = [];
       };
+
+      tlsSettings = mkOption {
+        description = ''
+          Make the server communicat via TLS.
+          This means it will only connect to other gluster
+          servers having certificates signed by the same CA.
+
+          Enabling this will create a file <filename>/var/lib/glusterd/secure-access</filename>.
+          Disabling will delete this file again.
+        '';
+        default = null;
+        type = types.nullOr (types.submodule {
+          options = {
+            tlsKey = mkOption {
+              default = null;
```

Comment:
I think you need to remove this `default = null` and the other two below.

Either we want that the entire `tlsSettings` dict is `null`, or we want that it is not null, and in that case we want that the user provides all 3 of `tlsKey`, `tlsPem` and `caCert`, so those should not be nullable.